ext/AdvancedHMCADTypesExt.jl

```diff
@@ -0,0 +1,26 @@
+module AdvancedHMCADTypesExt
+
+using AdvancedHMC: AbstractMetric, Hamiltonian
+using AbstractMCMC: LogDensityModel
+using LogDensityProblems: LogDensityProblems
+using LogDensityProblemsAD: LogDensityProblemsAD
+using ADTypes: AbstractADType
+
+function Hamiltonian(
+    metric::AbstractMetric, ℓπ::LogDensityModel, kind::AbstractADType; kwargs...
+)
+    return Hamiltonian(metric, ℓπ.logdensity, kind; kwargs...)
+end
+function Hamiltonian(metric::AbstractMetric, ℓπ, kind::AbstractADType; kwargs...)
+    if LogDensityProblems.capabilities(ℓπ) === nothing
+        throw(
+            ArgumentError(
+                "The log density function does not support the LogDensityProblems.jl interface",
+            ),
+        )
+    end
+    ℓ = LogDensityProblemsAD.ADgradient(kind, ℓπ; kwargs...)
```

Comment:
Let's use `DifferentiationInterface` to stay consistent with the rest of the TuringLang ecosystem. You could try to use `AbstractMCMC.LogDensityModel` and define `LogDensityProblem` interface on it in a similar way to [`DynamicPPL.LogDensityFunction`](https://github.com/TuringLang/DynamicPPL.jl/blob/main/src/logdensityfunction.jl).

Reply:
Yes, I am aware of that. DynamicPPL has dropped `LogDensityProblemsAD` to use `DifferentiationInterface` directly at suggestions by @willtebbutt

Reply:
Now DifferentiationInterface is supported